src/map/ai/states/item_state.cpp

```diff
@@ -0,0 +1,303 @@
+        m_PEntity->loc.zone->PushPacket(m_PEntity, CHAR_INRANGE_SELF, new CActionPacket(action));
+        Complete();
+    }
+    else if (IsCompleted() && tick > GetEntryTime() + m_castTime + std::chrono::milliseconds(m_PItem->getAnimationTime()))
+    {
+        m_PEntity->PAI->EventHandler.triggerListener("ITEM_STATE_EXIT", m_PEntity, m_PItem);
+        return true;
+    }
+    return false;
+}
+
+void CItemState::Cleanup(time_point tick)
+{
+    auto PChar = static_cast<CCharEntity*>(m_PEntity);
+
+    PChar->UContainer->Clean();
+
+    m_PItem->setSubType(ITEM_UNLOCKED);
+    PChar->pushPacket(new CInventoryFinishPacket());
+}
+
+bool CItemState::CanChangeState()
+{
+    return false;
+}
+
+void CItemState::TryInterrupt(CBattleEntity * PTarget)
+{
+    // todo: interrupt on being hit
+
+    if (PTarget)
+        PTarget = m_PEntity->IsValidTarget(PTarget->targid, m_PItem->getValidTarget(), m_errorMsg);
+    else
+        PTarget = m_PEntity->IsValidTarget(m_targid, m_PItem->getValidTarget(), m_errorMsg);
+
+    uint16 msg = 445; // you cannot use items at this time
+
+    if (HasMoved() || m_PEntity->StatusEffectContainer->HasPreventActionEffect())
+    {
+        m_interrupted = true;
+    }
+    else if (battleutils::IsParalyzed(m_PEntity))
+    {
+        msg = MSGBASIC_IS_PARALYZED;
+        m_interrupted = true;
+    }
+    else if (!PTarget)
+    {
+        m_interrupted = true;
+    }
+    else if (battleutils::IsIntimidated(m_PEntity, static_cast<CBattleEntity*>(PTarget)))
+    {
+        msg = MSGBASIC_IS_INTIMIDATED;
+        m_interrupted = true;
+    }
+
+    if (m_interrupted && !m_errorMsg)
+        m_errorMsg = std::make_unique<CMessageBasicPacket>(m_PEntity, m_PEntity, m_PItem->getID(), 0, msg);
+}
+
+CItemUsable* CItemState::GetItem()
+{
+    return m_PItem;
+}
+
+void CItemState::InterruptItem(action_t& action)
+{
+    TryInterrupt(dynamic_cast<CBattleEntity*>(GetTarget()));
+
+    if (m_interrupted)
+    {
+        action.id = m_PEntity->id;
+        action.actiontype = ACTION_ITEM_INTERRUPT;
+
+        actionList_t& actionList = action.getNewActionList();
+        actionList.ActionTargetID = (m_PEntity->IsValidTarget(m_targid, m_PItem->getValidTarget(), m_errorMsg) ? GetTarget()->id : 0);
+
+        actionTarget_t& actionTarget = actionList.getNewActionTarget();
+
+        actionTarget.reaction = REACTION_NONE;
+        actionTarget.speceffect = SPECEFFECT_NONE;
+        actionTarget.animation = 54;
+        actionTarget.param = 0;
+        actionTarget.messageID = 0;
+        actionTarget.knockback = 0;
+
+        auto PChar = static_cast<CCharEntity*>(m_PEntity);
+        PChar->pushPacket(m_errorMsg.release());
+
+        PChar->pushPacket(new CInventoryAssignPacket(m_PItem, INV_NORMAL));
+        PChar->pushPacket(new CInventoryItemPacket(m_PItem, m_PItem->getLocationID(), m_PItem->getSlotID()));
+    }
+}
+
+void CItemState::FinishItem(action_t& action)
```

Comment:
this function should be in charentity with the rest of the state callbacks (in case we ever need it to be polymorphic)